fontFeatures/feeLib/Chain.py

```diff
@@ -12,33 +12,60 @@
 """
 
 import fontFeatures
+from .Substitute import BASE_GRAMMAR, Substitute_GRAMMAR, Substitute
+from .util import extend_args_until
 
-GRAMMAR = """
-Chain_Args = normal_chain_args | contextual_chain_args
-
-normal_chain_args = gsluws+:stuff languages?:languages -> (stuff,languages, [], [])
-contextual_chain_args =  gsws*:pre '{' gsluws+:stuff languages?:languages '}' ws gsws*:post -> (stuff,languages, pre, post)
-gsluws = glyphselector:g ws lookup_list?:l ws -> (g,l)
-lookup_list = '(' ws lookups:l ws ')' -> l
-lookups = lookup*
-lookup = barename:b ws? ','? -> b["barename"]
-languages = '<' lang '/' script (ws ',' ws lang '/' script)* '>' ws
-lang = letter{3,4} | '*' # Fix later
-script = letter{3,4} | '*' # Fix later
+PARSEOPTS = dict(use_helpers=True)
+
+Chain_GRAMMAR = Substitute_GRAMMAR
+
+GRAMMAR = BASE_GRAMMAR+"""
+lookup: "^" BARENAME ","?
+lookups: lookup*
+gslu: glyphselector lookups
+
+normal_action: gslu+ languages?
+contextual_action: pre "(" normal_action ")" post
 """
 
 VERBS = ["Chain"]
 
-class Chain:
-    @classmethod
-    def action(self, parser, stuff, languages, pre, post):
-        inputs  = [x[0].resolve(parser.fontfeatures, parser.font) for x in stuff]
+class Chain(Substitute):
+    def lookups(self, args):
```

Comment:
Just putting a pin in this and if I see another one like it, I'm going to suggest we either provide a `passthrough` method we can assign methods to (`lookups = passthrough`), or fiddle with `__getattr__` to pass through any unknowns.